libr/cons/cons.c

```diff
@@ -2146,6 +2146,42 @@ R_API void r_cons_cmd_help(const char *help[], bool use_color) {
 	}
 }
 
+/* See r_cons_cmd_help().
+ * This version will only print help for a specific command.
+ * Will append spec to cmd before looking for a match, if spec != 0.
+ */
+R_API void r_cons_cmd_help_match(const char *help[], bool use_color, R_BORROW R_NONNULL char *cmd, char spec) {
+	const char **match = NULL;
+	const char *match_help_text[4];
+	int i;
+
+	if (spec) {
+		/* We now own cmd */
+		cmd = r_str_newf ("%s%c", cmd, spec);
+	}
+
+	for (i = 0; help[i]; i += 3) {
+		if (!strcmp (help[i], cmd)) {
+			match = &help[i];
+			break;
+		}
+	}
+
+	/* Command was not found in help text. */
+	r_return_if_fail (match);
+
+	/* Manually construct help array. No need to strdup, just borrow. */
+	match_help_text[3] = NULL;
+	for (i = 0; i < 3; i++) {
+		match_help_text[i] = match[i];
+	}
+	r_cons_cmd_help (match_help_text, use_color);
+
+	if (spec) {
+		free (cmd);
+	}
+}
```

Comment:
maybe return a pointer to the help string if any or null?

Reply:
do you mean `return match`? im fine with changing that, its only void bc r_core_cmd_help() returns void too.

right now this function considers no match to be a programming error but now that i think of it this causes problems if you dont check spec first, so that should be changed. i can change the return to char ** and the r_return_val to `return NULL`;

Reply:
not sure what the use case for a return value is

Reply:
Use the api somewhere so I can see an example usage